python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -942,6 +942,10 @@ def cron_schedule_from_schedule_type_and_offsets(
 class PartitionsSubset(ABC, Generic[T_str]):
     """Represents a subset of the partitions within a PartitionsDefinition."""
 
+    @property
+    def is_empty(self) -> bool:
+        return len(list(self.get_partition_keys())) == 0
```

Comment:
looks like here we still do the expensive thing, should we `cached_property` ? is there already caching in `get_partition_keys`? 
re: https://github.com/dagster-io/internal/discussions/9541
should this be `calculate_is_empty()` if it will continue to be costly?

Reply:
This is just for the base class, and so this implementation is not actually expensive for the DefaultPartitionsSubset class (or really any class other than TimeWindowPartitionsSubset). TimeWindowPartitionSubset has a fast implementation for this now, so I think this is ok naming-wise (and we should stick by the convention that this property will always be fast for future implementations)